edge/services/fft_client/horizon/pattern.json

```diff
@@ -0,0 +1,28 @@
+{
+  "name": "pattern-${SERVICE_NAME}",
+  "label": "FFT audio processing services",
+  "description": "Pattern for FFT audio processing services.",
+  "services": [
+    {
+      "serviceUrl": "${SERVICE_NAME}",
+      "serviceOrgid": "${HZN_ORG_ID}",
+      "serviceArch": "${ARCH}",
+      "serviceVersions": [
+        {
+          "version": "${SERVICE_VERSION}"
+        }
+      ]
+    },
+    {
+      "serviceUrl": "ibm.fft-server",
```

Comment:
This pattern causes the client and server services to be running on the same node. Is that required?

Reply:
I'd need some input from Vlad here, but when I tested via `hzn dev service start -S` and was watching the `mqtt` logs I realized that both the client and server were not running and I was not seeing any output at all. Only after I started both the client and server did I see the following
```
{"Timestamp":"2020-08-04T01:58:25Z","Trigger":false}
{"Timestamp":"2020-08-04T01:58:28Z","Trigger":true}
```
So I'm thinking they are both needed, but neither the client or server list the other as a required service, so he's making both to run with the pattern. I could be completely wrong though haha so if this is the case Vlad I think you could just make the server `required` by client and that would make them both run on the node as well.